contrib/kube-prometheus/jsonnet/kube-prometheus/prometheus/prometheus.libsonnet

```diff
@@ -166,8 +166,8 @@ local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
           version: $._config.versions.prometheus,
           baseImage: $._config.imageRepos.prometheus,
           serviceAccountName: 'prometheus-' + $._config.prometheus.name,
-          serviceMonitorSelector: selector.withMatchExpressions({ key: 'k8s-app', operator: 'Exists' }),
-          serviceMonitorNamespaceSelector: selector.withMatchExpressions({}),
+          serviceMonitorSelector: {},
```

Comment:
It looks like this line is not directly related to the underlying issue. Do we want this change?